### PR TITLE
Refactoring of Medias parameter

### PR DESCRIPTION
**Pull request by KvnMrs**
Issue: https://github.com/blockframes/blockframes/issues/7893

- [x] last remaining task : Update Excel files

### CODE REVIEW

File: libs/devops/src/lib/firestoreMigrations/0079.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      medias.splice(indexVideo, 1);
+    }
+
+    // rename media 'Planes' to Inflight
+    const indexPlanes = medias.indexOf('planes');
+    if (medias.includes('planes')) {
+      medias.splice(indexPlanes, 1, 'inflight');
+    }
+
+    term.medias = medias;
```
Comment (RemcoSimonides) [inline libs/devops/src/lib/firestoreMigrations/0079.ts:32]:
```
// remove media 'video'
term.medias = term.medias.filter(media => media !== 'video');
// rename media 'Planes' to 'Inflight'
term.medias = term.medias.map(media => media === 'planes' ? 'inflight' : media);
```

File: libs/devops/src/lib/firestoreMigrations/0079.ts [hunk 2]
```diff
@@ -42,25 +33,18 @@ async function updateBucketsMedias(db: Firestore) {
...
-          media.splice(indexPlanes, 1, 'inflight');
-        }
-        // remove media 'video'
-        if (media.includes('video')) {
-          const indexVideo = media.indexOf('video');
-          media.splice(indexVideo, 1);
-        }
-        return media;
+      bucket.contracts = bucket.contracts.map(contract => {
+        contract.terms[0].medias = contract.terms[0].medias
```
Comment (RemcoSimonides) [inline libs/devops/src/lib/firestoreMigrations/0079.ts:40]: I think you should do this for all the terms, not just the first.

File: libs/devops/src/lib/firestoreMigrations/0079.ts [hunk 3]
```diff
@@ -42,25 +33,18 @@ async function updateBucketsMedias(db: Firestore) {
   return runChunks(buckets.docs, async (doc) => {
     const bucket = doc.data();
     const contracts = bucket.contracts;
+
     // check if bucket has contracts
     if (contracts.length > 0) {
```
Comment (RemcoSimonides) [inline libs/devops/src/lib/firestoreMigrations/0079.ts:38]: I don't think this check is needed. If length is 0 then a function like `contracts.map()` wouldn't do anything.

File: libs/devops/src/lib/firestoreMigrations/0079.ts (hunk 2 above)
Comment (KvnMrs) [reply libs/devops/src/lib/firestoreMigrations/0079.ts]: all `terms `are composed of an index `0` and a single key `titleId`.

Capture

File: libs/devops/src/lib/firestoreMigrations/0079.ts [hunk 4]
```diff
@@ -32,19 +32,16 @@ async function updateBucketsMedias(db: Firestore) {
...
-          // rename media 'Planes' to Inflight
-          .map(media => media === 'planes' ? 'inflight' : media)
-          // remove media 'video'
-          .filter(media => media !== 'video');
-        return contract;
-      })
-    }
+
+    bucket.contracts = bucket.contracts.map(contract => {
+      contract.terms[0].medias = contract.terms[0].medias
```
Comment (RemcoSimonides) [inline libs/devops/src/lib/firestoreMigrations/0079.ts]:
```
      contract.terms.map(term =>
        term.medias = term.medias
                  // rename media 'Planes' to Inflight
                  .map(media => media === 'planes' ? 'inflight' : media)
                  // remove media 'video'
                  .filter(media => media !== 'video');
                return term; )
              return contract; )
```

**Review by RemcoSimonides — APPROVED** [review]
LGTM :)

**Pull request merged**